specs/CookieManagement.md

```diff
@@ -0,0 +1,505 @@
+# Background
+
+Cookie management in WebView has been one of the top feature requests. With that, the WebView2 team has introduced a new set of APIs allowing end developers to achieve goals such as authenticating the webview session, or retrieving cookies from webview to authenticate other tools.
+
+# Description
+
+One can get the `CookieManager` associated with a WebView to `GetCookies` via a `CookieList` collection, `AddOrUpdateCookie`, `DeleteCookies`, and `DeleteAllCookies` in a WebView environment.
+
+# Examples
+
+The following code snippet demonstrates how the cookie management APIs can be use:
+
+## Win32 C++
+
+```cpp
+ScenarioCookieManagement::ScenarioCookieManagement(AppWindow* appWindow)
+    : m_appWindow(appWindow), m_webView(appWindow->GetWebView())
+{
+    //! [CookieManager]
+    CHECK_FAILURE(m_webView->get_CookieManager(&m_cookieManager));
+    //! [CookieManager]
+
+    CHECK_FAILURE(m_webView->add_WebMessageReceived(
+        Microsoft::WRL::Callback<ICoreWebView2WebMessageReceivedEventHandler>(
+            [this](ICoreWebView2* sender, ICoreWebView2WebMessageReceivedEventArgs* args) {
+                wil::unique_cotaskmem_string uri;
+                CHECK_FAILURE(args->get_Source(&uri));
+
+                // Always validate that the origin of the message is what you expect.
+                if (uri.get() != m_sampleUri)
+                {
+                    return S_OK;
+                }
+                wil::unique_cotaskmem_string messageRaw;
+                CHECK_FAILURE(args->TryGetWebMessageAsString(&messageRaw));
+                std::wstring message = messageRaw.get();
+                std::wstring reply;
+
+                if (message.compare(0, 11, L"GetCookies ") == 0)
+                {
+                    GetCookiesHelper(message.substr(11));
+                    reply =
+                        L"{\"CookiesGot\":\"" + GetCookiesHelper(message.substr(11)) + L"\"}";
+                    CHECK_FAILURE(sender->PostWebMessageAsJson(reply.c_str()));
+                }
+                else if (message.compare(0, 10, L"AddOrUpdateCookie ") == 0)
+                {
+                    message = message.substr(10);
+                    std::wstring name = message.substr(0, message.find(' '));
+                    std::wstring value = message.substr(message.find(' ') + 1);
+
+                    //! [AddOrUpdateCookie]
+                    wil::com_ptr<ICoreWebView2Cookie> cookie;
+                    CHECK_FAILURE(m_cookieManager->CreateCookie(
+                        name.c_str(), value.c_str(), L".bing.com", L"/", &cookie));
+                    CHECK_FAILURE(m_cookieManager->AddOrUpdateCookie(cookie.get()));
+                    reply = L"{\"CookieAddedOrUpdated\":\"Cookie added or updated successfully.\"}";

```

Comment:
Recommend: change this to check for failure and message to client. Maybe just update macro name and add comment that its signaling the html in case of failure.

Reply:
Similarly for CookieToJsonString